### DISCUSSION TITLE
does this tool use an outside linter when --lint is requested?

### DISCUSSION THREAD

**Discussion by carl-hostrander** [aws/aws-sam-cli#5137] [Questions] [answered]
Hi! I have installed latest release of 1.82.0.   When I run 'sam validate --lint'  I am getting this:

You must specify a valid value for Runtime (java17). Valid values are ["dotnet6", "dotnetcore1.0", "dotnetcore2.0", "dotnetcore2.1", "dotnetcore3.1", "go1.x", "java11", "java8", "java8.al2", "nodejs", "nodejs10.x", "nodejs12.x", "nodejs14.x", "nodejs16.x", "nodejs18.x", "nodejs4.3", "nodejs4.3-edge", "nodejs6.10", "nodejs8.10", "provided", "provided.al2", "python2.7", "python3.10", "python3.6", "python3.7", "python3.8", "python3.9", "ruby2.5", "ruby2.7"]

I can deploy and run my lambda using java17.  Just wondering about this linting issue?  Does this tool use a 3rd party linter?

Thanks in advance.

**Comment by sriram-mv** [comment, marked as answer]
`sam validate --lint` uses `cfn-lint` under the hood.
